packages/rocketchat-mapview/client/mapview.coffee

```diff
@@ -10,15 +10,17 @@ class MapView
 		mv_enabled = RocketChat.settings.get 'MapView_Enabled'
 		mv_googlekey = RocketChat.settings.get 'MapView_GMapsAPIKey'
 
-		if message.location and mv_enabled
+		if message.location
 
 			# GeoJSON is reversed - ie. [lng, lat]
 			longitude = message.location.coordinates[0]
 			latitude = message.location.coordinates[1]
 
 			# confirm we have an api key set, and generate the html required for the mapview
-			if mv_googlekey?.length
+			if mv_enabled and mv_googlekey?.length
 				message.html  = '<a href="https://maps.google.com/maps?daddr='+latitude+','+longitude+'" target="_blank"><img src="https://maps.googleapis.com/maps/api/staticmap?zoom=14&size=250x250&markers=color:gray%7Clabel:%7C'+latitude+','+longitude+'&key='+mv_googlekey+'" /></a>'
+			else
+				message.html  = '<a href="https://maps.google.com/maps?daddr='+latitude+','+longitude+'" target="_blank">Shared Location</a>'
```

Comment:
Can you replace the `Shared Location` by `TAPi18n.__('Shared_Location')` and add the key to `en.i18n.json`?